[DO NOT MERGE] Reorganised Module Version Branch

Stringy opened the pull request:
Falco fork reorg branch, contains only Stackrox patches.

The upstream-main branch tracks upstream/master directly, and should not be modified (so do not merge this branch)
This branch can be renamed to module-version-2.5.0 when we're happy with the approach, and will form the basis for subsequent module versions.

The approach to updates will now be:

```
$ git checkout upstream-main
$ git fetch upstream && git rebase upstream/master
$ git checkout module-version-<current>
$ git checkout -b module-version-<new>
$ git rebase upstream-main
```

Molter73 committed:
Apply Stackrox changes to falcosecurity libs (#1)

Stringy committed:
Puts exe_writable logic behind ifdef guard for only newer kernels to enable backwards compatibility with kernels older than 4.9.0

keyallis committed:
Merge pull request #12 from stackrox/coc-file

Add CoC file with link

robbycochran committed:
Remove -g from bpf compilation (#24)

Molter73 committed:
Minor fixes after pulling changes from upstream

Molter73 committed:
Prefer static re2 library

Molter73 committed:
Double default ringbuffer size

Molter73 committed:
Also double the default ring buffer on kmod side

Stringy committed:
[Stackrox] Remove uname checks in driver builds (#41)

* Removes uname checks in driver builds
* Removes uname checks in driver bpf builds

Stringy committed:
[Stackrox] Reinstates double-free protection (#42)

Co-authored-by: Mauro Ezequiel Moltrasio <mmoltras@redhat.com>

erthalion committed:
Silence missing BPF program error (#40)

It's not in fact an error, as we need to exclude certain progs from the
probe. The present of a loud error can be confusing and interfere with
troubleshooting.

Stringy committed:
ppc64le fallback for syscall table values

Stringy committed:
open_by_handle_at use val_to_ring

Molter73 commented on CMakeLists.txt:
```diff
@@ -100,6 +101,12 @@ if (BUILD_SHARED_LIBS)
 	message(STATUS "Shared library soversion: ${FALCOSECURITY_SHARED_LIBS_SOVERSION}")
 endif()
 
+if(NOT WIN32 AND NOT APPLE)
+	option(NO_LINK_GRPC_LIBS "Do not link gRPC libraries" OFF)
+endif()
```
These 3 lines can easily be moved to collector's repo, since we don't support Windows or MacOS, we should be able to just set `NO_LINK_GRPC_LIBS` to `OFF`.

Stringy replied:
this is actually already forced to OFF in `collector/CMakeLists.txt` so I'll remove this.

Molter73 commented on cmake/modules/grpc.cmake:
We should probably consider rolling back the whitespace changes in this file, they are bound to create conflicts.

Molter73 commented on driver/CMakeLists.txt:
```diff
@@ -137,12 +117,16 @@ endif()
 # http://public.kitware.com/Bug/view.php?id=8438
 if(BUILD_DRIVER)
 	add_custom_target(driver ALL
+		COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/src/Makefile ${CMAKE_CURRENT_SOURCE_DIR}/Makefile
+		COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/src/driver_config.h ${CMAKE_CURRENT_SOURCE_DIR}/driver_config.h
```
At some point we need to consider removing these 2 lines, we have a similar copy in `prepare-src`:
https://github.com/stackrox/collector/blob/495eca30a96996f52636427de7e698ba56f37a08/kernel-modules/build/prepare-src#L93-L95
https://github.com/stackrox/collector/blob/495eca30a96996f52636427de7e698ba56f37a08/kernel-modules/build/prepare-src#L99-L100

Molter73 commented on driver/CMakeLists.txt:
```diff
@@ -137,12 +117,16 @@ endif()
 		VERBATIM)
 else()
 	add_custom_target(driver
+		COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/src/Makefile ${CMAKE_CURRENT_SOURCE_DIR}/Makefile
+		COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/src/driver_config.h ${CMAKE_CURRENT_SOURCE_DIR}/driver_config.h
```
Same as the previous comment.

Molter73 commented on driver/bpf/fillers.h:
```diff
@@ -3170,7 +3280,7 @@ FILLER(sys_open_by_handle_at_x, true)
 	/* Parameter 3: flags (type: PT_FLAGS32) */
 	u32 flags = (u32)bpf_syscall_get_argument(data, 2);
-	res = bpf_push_u32_to_ring(data, open_flags_to_scap(flags));
+	res = bpf_val_to_ring(data, open_flags_to_scap(flags));
```
Same as the previous comment.

Molter73 commented on driver/flags_table.c:
```diff
@@ -646,6 +646,11 @@ const struct ppm_name_value epoll_create1_flags[] = {
 	{0, 0},
 };
 
+const struct ppm_name_value machine_info_flags[] = {
+	{"BPF_STATS_ENABLED", PPM_BPF_STATS_ENABLED},
+	{0, 0},
+};
```
I'm not entirely sure where this is coming from, it's not in our master or the upstream master, think it might be an old change that was removed upstream and somehow crawled its way back in.

Stringy replied:
removed

Molter73 commented on driver/ppm_events_public.h:
```diff
@@ -2180,4 +2180,9 @@ struct ppm_event_entry {
+/*
+ * machine_info flags.
+ */
+#define PPM_BPF_STATS_ENABLED		(1 << 0)
```
Same as last comment.

Stringy replied:
removed

Molter73 commented on userspace/libsinsp/parsers.cpp:
There are some whitespace changes in this file that we might want to drop.

Stringy replied:
fixed

Stringy replied:
damn it, I was sure they were gone. I'll fix it.

Stringy commented on cmake/modules/CompilerFlags.cmake:
```diff
@@ -75,3 +75,7 @@ else() # MSVC
 	add_definitions(-D_HAS_STD_BYTE=0)
 endif()
+
+if(APPLE)
+	set(CMAKE_EXE_LINKER_FLAGS "-pagezero_size 10000 -image_base 100000000")
+endif()
```
Will remove

Stringy commented on driver/bpf/fillers.h:
```diff
@@ -3156,7 +3266,7 @@ FILLER(sys_open_by_handle_at_x, true)
 	/* Parameter 1: ret (type: PT_FD) */
 	long retval = bpf_syscall_get_retval(data->ctx);
-	int res = bpf_push_s64_to_ring(data, retval);
+	int res = bpf_val_to_ring(data, (s64)retval);
```
Unfortunately using the upstream way causes verifier errors. We don't ever call this filler, so I think as long as we can satisfy the verifier, we should be ok. Having said that, it might be worth a comment here that describes why we can't use the new type-specific functions

Molter73 replied:
That's good to know. They specifically created this alternative function to deal with verifier errors so it sucks that it's doing the opposite, but validates my objection to it a couple months back.

Stringy commented on cmake/modules/libsinsp.cmake:
```diff
@@ -75,12 +75,22 @@ function(set_sinsp_target_properties target)
+    add_subdirectory(${LIBSINSP_DIR}/userspace/libsinsp ${PROJECT_BINARY_DIR}/libsinsp)
+endif() # BUILD_USERSPACE
+
+set(LIBSINSP_LIB "${PROJECT_BINARY_DIR}/libsinsp/libsinsp.a")
+install(FILES "${LIBSINSP_LIB}" DESTINATION "${CMAKE_INSTALL_LIBDIR}/${LIBS_PACKAGE_NAME}"
+			COMPONENT "sinsp")
```
it was unrelated, so I've removed it.

Molter73 commented on CMakeLists.txt:
```diff
@@ -1,3 +1,4 @@
+
```
[ultra-nit] Why the extra line?

Molter73 reviewed (approved):
LGTM!

kcrane committed:
ebpf probe changes for s390x enablement (#44)

* fix(collector_probe): align sys exit format member size

The generic syscall format uses long (8 byte) for the syscall id
in the sys exit format.  Attaching directly to syscalls, their
sys exit format uses an int (4 bytes).  Correct this size when
direct-attached probes are used to prevent reading at the right
member offset (in falco fillers).

Signed-off-by: Hendrik Brueckner <brueckner@de.ibm.com>

* fix(collector): disable CAPTURE_SCHED_PROC_FORK on s390

CAPTURE_SCHED_PROC_FORK requires raw trace point support/format which
does not work with collector custom probe design.

Signed-off-by: Hendrik Brueckner <brueckner@de.ibm.com>

* fix(bpf): corrected merge for bpf probe reads

Signed-off-by: Hendrik Brueckner <brueckner@de.ibm.com>

* update(bpf): socketcall support for regular trace points

Signed-off-by: Hendrik Brueckner <brueckner@de.ibm.com>

* Create helper to convert socketcall to network calls based on socketcall id.  Needed because socketcall args are not yet stored in BPF argument map on enter probe

* require USE_BPF_PROBE_KERNEL_USER_VARIANTS on s390x

* remove USE_COLLECTOR_CUSTOM_PROBES

---------

Signed-off-by: Hendrik Brueckner <brueckner@de.ibm.com>
Co-authored-by: Hendrik Brueckner <brueckner@linux.ibm.com>
Co-authored-by: Hendrik Brueckner <brueckner@de.ibm.com>
Co-authored-by: Kirk Crane <kcrane@redhat.com>
